### PR TITLE
[WIP] :building_construction:  support typescript code

### PR DESCRIPTION
**Pull request by MatissJanis**
A few changes in this PR to facilitate migration to TypeScript.

1. Tooling setup to allow developing with TS files
2. Changing imports/exports to be more modern
3. Changing some file paths now that files are served from `./build` folder instead of `.`

**Why do this?**

Upgrading some dependencies with security issues (such as node-fetch) requires using modern import syntax. So we might as well continue the TS migration here..

### CODE REVIEW

**Comment by MatissJanis** [comment]
Note: I did not yet test actually deploying the build to pikapods/fly.io or anywhere else. Need to check if this still works as expected before merging.

**Comment by MatissJanis** [comment]
Also: maybe this is an overkill. Still need to consider alternatives.

But either way we need to figure out the import issue: https://github.com/node-fetch/node-fetch/blob/HEAD/docs/v3-UPGRADE-GUIDE.md#converted-to-es-module

File: app.js [hunk 1]
```diff
@@ -35,12 +36,19 @@ app.use((req, res, next) => {
...
 });
 app.use(
-  express.static(__dirname + '/node_modules/@actual-app/web/build', {
-    index: false
-  })
+  express.static(
+    path.resolve(__dirname + '/../node_modules/@actual-app/web/build'),
+    {
+      index: false
+    }
```
Comment (j-f1) [inline app.js:43]:
```suggestion
    { index: false }
```

File: app.js [hunk 2]
```diff
@@ -35,12 +36,19 @@ app.use((req, res, next) => {
...
+  )
 );
 app.get('/*', (req, res) => {
-  res.sendFile(__dirname + '/node_modules/@actual-app/web/build/index.html');
+  res.sendFile(
+    path.resolve(
+      __dirname + '/../node_modules/@actual-app/web/build/index.html'
+    )
+  );
```
Comment (j-f1) [inline app.js:51]: Looking at the check failure (which is probably safe to ignore like you did), we could pull the file out into a top level `fs.readFileSync` and then just send back the buffer each time.

File: jest.setup.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const fs = require('fs');
 const { join } = require('path');
 const { getAccountDb } = require('./account-db');
-const config = require('./load-config');
+const { default: config } = require('./load-config');
```
Comment (j-f1) [inline jest.setup.js:4]: Is it possible to switch this file to ESM too? real ESM files can’t be `require()`-ed synchronously.

**Review by j-f1 — APPROVED** [review]
This would be cool! I have a couple of specific thoughts below. Also, IMO we should upgrade to Node 19 (I was just getting started on that but hadn’t actually done anything yet) so that we can use the `fetch()` global and other newer features/security improvements.

**Review by j-f1 — COMMENTED** [review]
oops wrong button

**Pull request closed without merging**